framework/src/transfers/MultiAppCopyTransfer.C

```diff
@@ -51,204 +52,99 @@ MultiAppCopyTransfer::initialSetup()
 }
 
 void
-MultiAppCopyTransfer::execute()
+MultiAppCopyTransfer::transferDofObject()
 {
-  _console << "Beginning CopyTransfer " << name() << std::endl;
-
-  switch (_direction)
-  {
-    case TO_MULTIAPP:
+  if (_to_object->n_dofs(_to_sys_num, _to_var_num) > 0) // If this variable has dofs at this node
+    for (unsigned int comp = 0; comp < _to_object->n_comp(_to_sys_num, _to_var_num); ++comp)
     {
-      FEProblem & from_problem = _multi_app->problem();
-      MooseVariable & from_var = from_problem.getVariable(0, _from_var_name);
-
-      MeshBase * from_mesh = NULL;
-
-      from_mesh = &from_problem.mesh().getMesh();
-
-      SystemBase & from_system_base = from_var.sys();
-
-      System & from_sys = from_system_base.system();
-      unsigned int from_sys_num = from_sys.number();
-
-      // Only works with a serialized mesh to transfer from!
-      mooseAssert(from_sys.get_mesh().is_serial(), "MultiAppCopyTransfer only works with ReplicatedMesh!");
-
-      unsigned int from_var_num = from_sys.variable_number(from_var.name());
-
-      //Create a serialized version of the solution vector
-//      NumericVector<Number> * serialized_solution = NumericVector<Number>::build(from_sys.comm()).release();
-//      serialized_solution->init(from_sys.n_dofs(), false, SERIAL);
-
-      // Need to pull down a full copy of this vector on every processor so we can get values in parallel
-//      from_sys.solution->localize(*serialized_solution);
-
-      for (unsigned int i=0; i<_multi_app->numGlobalApps(); i++)
-      {
-        if (_multi_app->hasLocalApp(i))
-        {
-          MPI_Comm swapped = Moose::swapLibMeshComm(_multi_app->comm());
-
-          // Loop over the master nodes and set the value of the variable
-          System * to_sys = find_sys(_multi_app->appProblem(i).es(), _to_var_name);
-
-          unsigned int sys_num = to_sys->number();
-          unsigned int var_num = to_sys->variable_number(_to_var_name);
+      dof_id_type dof = _to_object->dof_number(_to_sys_num, _to_var_num, comp);
 
-          NumericVector<Real> & solution = _multi_app->appTransferVector(i, _to_var_name);
+      Moose::swapLibMeshComm(_swapped);
```

Comment:
I actually think we can remove all of this comm swapping (ahem).

The only reason to do it now is for the solve (because it swaps PETSc parameters).  It's unnecessary for normal stuff now (because all object have their own comm implicitly).

Anyway - that's not a call to action... just noticing it.